src/mapping/create.ts

```diff
@@ -0,0 +1,67 @@
+import { type DataHandlerContext, type Log } from "@subsquid/evm-processor";
+
+import * as spec from "../abi/create";
+import { type Fields, type NETWORK } from "../consts";
+import { type Store } from "../db";
+import { EntityBuffer } from "../entityBuffer";
+import { RentalStarted } from "../model";
+import { randomUUID } from "crypto";
+
+export function parseEvent(
+  ctx: DataHandlerContext<Store>,
+  log: Log<Fields>,
+  chain: NETWORK,
+): void {
+  try {
+    switch (log.topics[0]) {
+      case spec.events.RentalOrderStarted.topic: {
+        const e = spec.events.RentalOrderStarted.decode(log);
```

Comment:
Instead of index access e[0] etc you could:
```
const [
  orderHash,
  emittedExtraData,
  seaportOrderHash,
  items, // might need a different name w/r semantics?
  hooks, // might need a different name w/r semantics?
  lender,
  renter,
  rentalWallet,
  endTimestamp,
] = spec.events.RentalOrderStarted.decode(log);
```
This might make things a bit more clear in the code below. If `items` and `hook` are correct terms you could consider inlining the `map()`s:

```
rentalStarted.items = items.map((item) => ({
  ...item,
  id: randomUUID(),
  rentalStarted, // note that you don't need colon notation if key/value are the same
}));

rentalStarted.hooks = items.map((hook) => ({
  ...hook,
  id: randomUUID(),
  rentalStarted,
}));
```

Also, random thought, because the model instantiation is from a `class`, you could consider moving this part inside the class itself on the constructor? 🤔  I'll post a suggestion below.